sqlmesh/core/engine_adapter/base.py

```diff
@@ -2115,6 +2137,11 @@ def _build_table_properties_exp(
                 )
             )
 
+        if table_properties:
+            table_type = self._extract_creatable_type_from_properties(table_properties)
+            if table_type:
+                properties.append(table_type)
```

Comment:
I think we may need to do something like this here instead (no need for branching):

```python
properties.extend(ensure_list(extract(...)))
```